README.md

```diff
@@ -1 +1,42 @@
-# narayana-proposals
\ No newline at end of file
+# Narayana Proposals
+
+## Introduction
+
+This workflow provides a guideline to formally define the stages to propose, design, implement, and merge new Requests for Enhancement (RFEs) in the Narayana project. The process defined here can be followed by any contributor who would like to reach pre-agreement for the design they would like to use for the implementation of RFEs in the Narayana project. On the other hand, the employment of this workflow is _highly recommended_ when addressing larger scoped or complex features, and when there are disagreements among the community members of Narayana. If contributors have any uncertainties about whether this workflow is appropriate for their RFE, we strongly recommend that they reach out to the team to determine whether their investigation area would benefit from sharing design documents before proceeding with the development of their solution. Failing to do so might result in the realisation that the design is incompatible with Narayana's architecture.
+
+The ultimate goal of this workflow is the optimisation of the proposal management of the Narayana project, in the effort of achieving a smoother transition of new RFEs from conceptualisation to realisation while fostering collaboration and knowledge sharing in our community.
+
+## 0. Secure the approval
+
+One of the fundamental improvements introduced by this workflow is the implementation of a process to secure the approval of the Narayana team (which consists of the [Contributors](https://www.narayana.io/community/index.html) in “Our team”, who are responsible for maintaining the Narayana code base). To progress a proposal, the whole Narayana team is required to approve it within a defined time window, which is set to 14 days\*. Optional members - specifically [Members](https://github.com/orgs/jbosstm/people?query=role%3Amember) and [Outside Contributors](https://github.com/orgs/jbosstm/outside-collaborators) - have the flexibility to participate at their discretion and their opinion is equal in importance to the [Narayana team](https://www.narayana.io/community/index.html). As a consequence, every contributor from the aforementioned groups (hereafter “reviewers”) can veto the approval of a proposal within the defined time window.
```

Comment:
just fyi, non collaborators can't open github links you have here

Reply:
Modifying the visibility of memberships is something outside our control, unfortunately :confused: .

Reply:
But you can control whether or not you include the link.

Reply:
I am not sure if GitHub can provide this? If it does not, we could consider to update https://www.narayana.io/community/index.html with the information and refer there instead. I think I favor that, but what do others think?

Reply:
I was talking about [this](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-membership-in-organizations/publicizing-or-hiding-organization-membership).

Reply:
Nice! But I do wonder if that list queryable? I didn't spot a way to search for people that are members of an organisation. I am wondering, do we need people who are not members to know who the members are, or just a way for people that are able to merge the pull request a way to check the reviews were from a member etc? If it's the latter we can just remove the links and use the definitions `[Members]` and `[Outside Contributors]` expecting that as they are Github ones, the people in control of merging should be able to find that information?

Reply:
> Nice! But I do wonder if that list queryable?

It is queryable indeed. I've try to change the visibility of my membership and I can see myself in the list of members even when I access `jbosstm` in a private window